src/Sarif.Multitool/Rules/SARIF2010.ProvideCodeSnippets.cs

```diff
@@ -65,5 +78,46 @@ private void AnalyzeRegion(Region region, string regionPointer)
                     nameof(RuleResources.SARIF2010_ProvideCodeSnippets_Note_Default_Text));
             }
         }
+
+        private bool AnalyzeArtifactLocation(ArtifactLocation artifactLocation)
+        {
+            // no artifactLocation / no artifacts, so we should look for the snippet
+            if (artifactLocation == null || this.artifacts == null)
+            {
+                return true;
+            }
+
+            artifactLocation.TryReconstructAbsoluteUri(this.originalUriBaseIds, out Uri resolvedUri);
```

Comment:
>TryReconstructAbsoluteUri [](start = 29, length = 25)

If a SARIF viewer can't resolve the URI, it won't be able to find the embedded content, and so it will need the snippet to be present. So:
```
if (!artifactLocation.TryReconstructAbsoluteUri(this.originalUriBaseIds, out Uri resolvedUri))
{
    return false;
}
```

We need a test case for this, so please take a look at `TryReconstructAbsoluteUri` to see how you can make it fail. Probably using an unknown `uriBaseId` will do it.

Reply:
yes and no...if we couldn't resolve doesn't mean that we need to look for snippet. Imagine the situation where we don't have the uriBaseId but we have the file. That file can be presented in the artifacts list, right?

---
In reply to: [462538548](https://github.com/microsoft/sarif-sdk/pull/2019#discussion_r462538548) [](ancestors = 462538548)

Reply:
You're right. Please add a comment, because people reading the code will be surprised (as I was) that you call a `Try` API without checking for success.

---
In reply to: [462546597](https://github.com/microsoft/sarif-sdk/pull/2019#discussion_r462546597) [](ancestors = 462546597,462538548)

Reply:
Added some comments, let me know what do you think

---
In reply to: [462575560](https://github.com/microsoft/sarif-sdk/pull/2019#discussion_r462575560) [](ancestors = 462575560,462546597,462538548)

Reply:
Good. Wording: "... since neither originalUriBaseIds nor artifactLocation.UriBaseId is required."

---
In reply to: [462582648](https://github.com/microsoft/sarif-sdk/pull/2019#discussion_r462582648) [](ancestors = 462582648,462575560,462546597,462538548)